docs: 添加csv表头优化相关TODO

A-n-uo opened the pull request:
进行csv格式优化

nan-mu committed:
docs: 添加csv表头优化相关TODO

nan-mu committed:
docs: 将文档迁移到内部仓库

nan-mu committed:
但是wtr相应部分还没完善

live3w committed:
大体修改了语法错误，有些错误不太懂

live3w committed:
fix(大体修改了错误): 大体修改了错误，有一些错误不太懂

nan-mu commented on 1.mmd:
这是啥？没用的话删了

live3w replied:
已经删除了

nan-mu commented on src/handler.rs:
```diff
@@ -15,9 +16,29 @@ pub enum Consumer {
 use log::{debug, error, info, warn};
 use tokio::sync::broadcast;
+
+// 定义 CsvRecordA 结构体
+#[derive(Debug, Serialize)]
+pub struct CsvRecordA {
+    //pub timestamp: String,  // 时间戳
+    pub mark: Option<String>, // 可选标记
+    pub channel: u8,          // 通道
+    pub data: u32,            // 数据
+}
+
+// 定义 CsvRecordB 结构体
+#[derive(Debug, Serialize)]
+pub struct CsvRecordB {
+    // pub timestamp: String,  // 时间戳
+    pub mark: Option<String>, // 可选标记
+    pub channel: u8,          // 通道
+    pub data: u32,            // 数据
+    pub settlecount: u16,     // settlecount 寄存器值
+    pub rcount: u16,          // rcount 寄存器值
+}
```
嗯...最好起一个有含义的名字
```suggestion
// 定义 CsvRecordA 结构体
#[derive(Debug, Serialize)]
pub struct CsvRecordA {
    //pub timestamp: String,  // 时间戳
    pub mark: Option<String>, // 可选标记
    pub channel: u8,          // 通道
    pub data: u32,            // 数据
}

// 定义 CsvRecordB 结构体
#[derive(Debug, Serialize)]
pub struct CsvRecordB {
    // pub timestamp: String,  // 时间戳
    pub mark: Option<String>, // 可选标记
    pub channel: u8,          // 通道
    pub data: u32,            // 数据
    pub settlecount: u16,     // settlecount 寄存器值
    pub rcount: u16,          // rcount 寄存器值
}
```

live3w replied:
> 嗯...最好起一个有含义的名字

嘶，这个名字就是定义的两种结构体ab的名字，没想好咋改

nan-mu commented:
@A-n-uo @live3w

nan-mu reviewed (changes requested):
可见性测试1

nan-mu reviewed (approved):
可见性测试2

nan-mu reviewed (commented):
可见性测试3

nan-mu commented on src/config.rs:
可见性测试4

这样，我们将RecordType从handler::a移到这里，并且改名为CsvRecordType。我认为这样语义更清晰一些，CsvRecordType只会包含表意的枚举类型，而在handler::a很明显以及有具体的csv表头结构题代表这些类型了。

再者，handler::a中只有代码使用，而config这里作为类型使用。类型声明和它的子类型应该靠近。

nan-mu commented on src/handler.rs:
```diff
@@ -15,9 +16,29 @@ pub enum Consumer {
 use ldc1614::Channel;
 use log::{debug, error, info, warn};
 use tokio::sync::broadcast;
+
+// 定义 CsvRecordA 结构体
+#[derive(Debug, Serialize)]
+pub struct CsvRecordA {
+    //pub timestamp: String,  // 时间戳
+    pub mark: Option<String>, // 可选标记
+    pub channel: u8,          // 通道
+    pub data: u32,            // 数据
+}
+
+// 定义 CsvRecordB 结构体
+#[derive(Debug, Serialize)]
+pub struct CsvRecordB {
+    // pub timestamp: String,  // 时间戳
+    pub mark: Option<String>, // 可选标记
+    pub channel: u8,          // 通道
+    pub data: u32,            // 数据
+    pub settlecount: u16,     // settlecount 寄存器值
+    pub rcount: u16,          // rcount 寄存器值
+}
```
可见性测试5
像之前说的，三个需求对应三个结构体。但现在看来这部分想在新的代码里面复用的话，我们可能就只需要两个结构体，分别是最简单的和最复杂的。前者只有position和data，后者除了这两者还要有channel，time，settlecount，rount，mark。

这部分脱了好久了，请尽快解决，起名困难就问ai，需求发给他。

nan-mu commented on src/handler.rs:
```diff
@@ -144,45 +189,91 @@ impl Handler {
+                                                });
+                                        }
+                                        Record::B(record_b) => {
+                                            // 处理 RecordB 类型的数据
+                                            let key = match record_b.channel {
+                                                ldc1614::Channel::Zero => "Channel0",
+                                                ldc1614::Channel::One => "Channel1",
+                                                ldc1614::Channel::Two => "Channel2",
+                                                ldc1614::Channel::Three => "Channel3",
+                                            };
+
+                                            // 处理 mark，如果有则加入到 key 中
+                                            let key = if let Some(ref string) = record_b.mark {
+                                                format!("{}:{}", key, string)
+                                            } else {
+                                                key.to_string()
+                                            };
+
+                                            // Redis 写入操作，可以使用 record_b 的数据
+                                            client
+                                                .ts_add::<usize, String, i64, RedisMap>(
+                                                    key.clone(),
+                                                    0, // 使用 0 或者替换为实际的 timestamp
+                                                    record_b.data as f64,
+                                                    None,
+                                                    None,
+                                                    None,
+                                                    None,
+                                                    RedisMap::new(),
+                                                )
+                                                .await
+                                                .unwrap_or_else(|err| {
+                                                    error!(
+                                                        "写入redis失败, 数据为: {:?}, 错误: {:?}",
+                                                        record_b, err
+                                                    );
+                                                    0 // 返回默认的 usize 值
+                                                });
+                                        }
+                                    }
```
可见性测试6

算是一个新任务，我建议你们新开一个分支。我需要删除所有和redis相关的代码，以防止像这样计划外的工作量。

新任务你们自己安排下人手，就从csv-perf差分再合并到csv-perf吧。和之前的任务一样，给我一个你们做好需要我做第一次审查的时间点。我很希望你们在这周末完成csv-perf和这个新任务

nan-mu commented on src/main.rs:
可见性测试7

我很好奇你们为什么没对main函数中的submit进行修改或者提出问题，在我看来这里缺少了两个东西：
1. 从yaml中读取配置，了解到你们应该使用何种csv文件格式
2. 对submit需要的参数做出隔离。现在的代码中通过直接传参让数据发往handler。但这种设计并不适合我们可变参数的场景。对于所有可能的参数，我们可以将一条csv记录需要的参数分为两类，一类是main函数提供的，一类是submit提供的。前者包括寄存器当前值，电机位置，mark（这个其实是预留参数，先不管，我本来的计划是从yaml里提供），数据来源通道。然后就是submit，其实只提供data和时间。对于这里的设计我希望你们先提提意见。但时间不要拖太长，这个任务已经拖太久了。

nan-mu reviewed (changes requested):
可见性测试8

nan-mu committed:
feat: 支持 CSV 文件记录类型配置和改进 matlab_type_range 函数

1. 修改 .gitignore 文件，添加对 .diff 文件的忽略。
2. 删除不必要的文件 1.mmd 和 a。
3. 更新 Cargo.toml 文件，调整依赖版本：
   - env_logger 版本从 0.11.5 改为 0.11
   - log 版本从 0.4.22 改为 0.4
   - 添加 once_cell 依赖，版本为 1.20
4. 在 config.rs 文件中：
   - 添加 RecordType 枚举类型，支持 Easy、WithRegister 和 All 三种记录类型。
   - 使用 once_cell::sync::OnceCell 来存储全局的 RecordType。
   - 修改 matlab_type_range 函数，支持逗号分隔的字符串，并返回 Vec<T>。
5. 在 handler.rs 文件中：
   - 删除 Redis 相关代码。
   - 修改 Handler 结构体和 submit 方法，支持新的 RecordType。
   - 修改 CsvRecordA 和 CsvRecordB 结构体，分别为 EasyRecord、RegRecord 和 AllRecord。
6. 在 main.rs 文件中：
   - 读取配置文件中的 record_type，并设置全局的 RECODE_TYPE。
   - 修改任务处理逻辑，支持新的 RecordType。
7. 更新任务配置文件 task-example.yml，添加 record_type 配置项。
8. 更新测试用例，验证 matlab_type_range 函数的新功能。

nan-mu committed:
test: 更新配置和测试，改进代理端口设置

1. 更新 README.md 文件：
   - 修改 rsync 命令，更新排除项和目标路径。

2. 修改 src/config.rs 文件：
   - 修改 matlab_type_range 函数，修正 while 循环条件，将 `<=` 改为 `<`。
   - 更新测试用例路径，修正配置文件路径。

3. 修改 src/motor.rs 文件：
   - 添加测试模块，使用 #[cfg(test)] 进行条件编译。
   - 添加 tokio 异步测试，验证 Motor 结构体的 goto 方法。
   - 添加 #[should_panic] 注解，验证异常情况。

4. 添加通过环境变量配置代理端口的说明：
   - 在 README.md 中添加如何通过环境变量设置代理端口的说明。
   - 提供示例脚本和命令，指导用户如何设置 ALL_PROXY、HTTP_PROXY 和 HTTPS_PROXY 环境变量。

nan-mu committed:
Merge branch 'dev' into csv-perf

nan-mu reviewed (approved):
合！
